logstash/datadog_checks/logstash/logstash.py

```diff
@@ -99,6 +108,13 @@ class LogstashCheck(AgentCheck):
         "logstash.pipeline.plugins.filters.events.duration_in_millis": ("gauge", "events.duration_in_millis"),
     }
 
+    def __init__(self, name, init_config, instances):
+        super(LogstashCheck, self).__init__(name, init_config, instances)
+
+        if not ('read_timeout' in self.instance or 'connect_timeout' in self.instance):
+            timeout = self.instance.get('timeout', self.DEFAULT_TIMEOUT)
+            self.http.options['timeout'] = (timeout, timeout)
+
```

Comment:
It's probably fine to remove that part and update the check to have a 10 seconds timeout like all other integrations.

This part adds extra complexity and doesn't work in all cases (if only read_timeout is set then connect_timeout is set to 10 seconds and not 5 like expected)